Source/Client/CMakeLists.txt

```
@@ -5,10 +5,63 @@ message(STATUS "Configuring Client")
 ###############################################################################
 # Configure Step
 ###############################################################################
-add_executable(Client
-    WIN32
-    "${PROJECT_SOURCE_DIR}/Resources/Shared/Icons/appicon.rc"
-)
+if (CMAKE_SYSTEM_NAME STREQUAL "Windows")
+    add_executable(Client
+        WIN32
+        "${PROJECT_SOURCE_DIR}/Resources/Shared/Icons/appicon.rc"
+    )
+elseif (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+    # Add the app icon to the app bundle
+    set(CLIENT_ICON "${PROJECT_SOURCE_DIR}/Resources/Shared/Icons/icon.icns")
+    set_source_files_properties(${CLIENT_ICON}
+                                PROPERTIES MACOSX_PACKAGE_LOCATION "Resources")
+
+    # Add the PkgInfo file to the app bundle
+    set(CLIENT_PKGINFO "${PROJECT_SOURCE_DIR}/Resources/Client/macOS/PkgInfo")
+    set_source_files_properties(${CLIENT_PKGINFO}
+                                PROPERTIES MACOSX_PACKAGE_LOCATION .)
+
+    add_executable(Client
+        MACOSX_BUNDLE
+        ${CLIENT_ICON}
+        ${CLIENT_PKGINFO}
+    )
+
+    # Reduce the amount of libraries bundled by telling the linker
+    # to remove references to libraries which aren't used.
+    target_link_options(
+        Client
+        PRIVATE
+        "LINKER:-dead_strip,-dead_strip_dylibs"
+    )
+
+    # Configure the Info.plist file
+    string(TIMESTAMP CurrentYear "%Y")
+    set(PROJECT_CREATOR_NAME "Net5F")

```

```suggestion
    set(PROJECT_CREATOR_NAME "Net_")
```
I couldn't get my usual name on github, but I like the underscore 😅